### PR TITLE
Upgrade Celery and remove some now-unnecessary cruft (bug 1194181)

### PR DESCRIPTION
**Pull request by kmaglione**
We're going to need to add these three packages to the wheelhouse and have Jason change the worker deploy scripts to launch `celery` directly rather than via `manage.py`.

### CODE REVIEW

File: apps/amo/celery.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+task = app.task
+
+app.config_from_object('django.conf:settings')
+app.autodiscover_tasks(settings.INSTALLED_APPS)
+
+
+@task_failure.connect
+def process_failure_signal(exception, traceback, sender, task_id,
+                           signal, args, kwargs, einfo, **kw):
+    exc_info = (type(exception), exception, traceback)
```
Comment (kmaglione) [inline apps/amo/celery.py]: This is a slightly modified version is what was previously imported from `celeryutils`. The rest of that module is no longer required.

**Comment by magopian** [comment]
This looks nice, thanks!

Let's wait for PR #673 (which this PR seems based on) to be landed.

**Comment by magopian** [comment]
#673 has landed, so now this needs a rebase. I think we discussed what should be done with the "do not merge" commit, but I don't remember, can you please restate it here?

**Comment by kmaglione** [comment]
Rebased and removed the 'DO NOT MERGE' commit, but I think we're going to need these new requirement versions in the pyrepo before it passes: https://github.com/mozilla/olympia/pull/678/files#diff-668c57ad76e35851be983271a1a9ac5bR3

**Comment by kmaglione** [comment]
Ah. Or not.

**Comment by kmaglione** [comment]
Just need @jasonthomas to update the dev servers to launch the Celery workers in the new way, then.

**Comment by magopian** [comment]
I'm really curious how the travis build could pass... without the packages updated on pyrepo?!
I've now uploaded all the versions (packages and wheels) to pyrepo.

**Comment by kmaglione** [comment]
They were apparently already there.

**Pull request merged**